experimental/segmenter/src/line.rs

```diff
@@ -124,11 +125,14 @@ impl LineBreakSegmenter {
             .load_resource(&DataRequest::default())?
             .take_payload()?;
 
-        // TODO: Use `provider` parameter after we support loading dictionary data from the
-        // production-ready providers.
-        let inv_provider = icu_provider::inv::InvariantDataProvider;
-        let dictionary_payload = inv_provider
-            .load_resource(&DataRequest::default())?
+        let locale: Locale = ("th")
+            .parse()
+            .map_err(|_| DataError::custom("cannot parse locale string"))?;
```

Comment:
Suggestion, here and elsewhere: use the macro (which expands to a const function)

```suggestion
        let locale = locale!("th");
```